### PR TITLE
[exporter/loki] Don't retry on 4xx responses from loki

### PR DESCRIPTION
**Pull request by gregoryfranklin**
**Description:** Don't retry when receiving a 4xx response (excluding 429) from loki.

**Link to tracking Issue:** #12930

### CODE REVIEW

File: exporter/lokiexporter/exporter.go [hunk 1]
```diff
@@ -150,6 +150,12 @@ func (l *lokiExporter) pushLogData(ctx context.Context, ld plog.Logs) error {
 			line = scanner.Text()
 		}
 		err = fmt.Errorf("HTTP %d %q: %s", resp.StatusCode, http.StatusText(resp.StatusCode), line)
+
+		// Errors with 4xx status code (excluding 429) should not be retried
+		if resp.StatusCode >= http.StatusBadRequest && resp.StatusCode < http.StatusTooManyRequests {
```
Comment (jpkrohling) [inline exporter/lokiexporter/exporter.go:155]: This is ruling out 431 and 451, which might be OK, as I'm confident Loki does not return 451 anyway, and it likely does not return 431 either.

I would reword the comment a bit, then. Otherwise, looks good.

Comment (gregoryfranklin) [reply exporter/lokiexporter/exporter.go]: 431 and 451 look like they should also be permanent so I've changed the if statement to actually cover all 4xx codes including those greater than 429 but still excluding 429.

ie. the code now does what the comment describes.

**Review by jpkrohling — APPROVED** [review]
